src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,151 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (size == 0) {
+            head = newNode;
+        } else {
+            tail.nextElement = newNode;
+        }
+        tail = newNode;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0 && index < size) {
+            Node<T> atIndexNode = head;
+            atIndexNode.prevElement = new Node<>(null, value, atIndexNode);
+            head = atIndexNode.prevElement;
+            size++;
+            return;
+        }
+        if (index > 0 && index < size) {
+            Node<T> atIndexNode = getNode(index);
+            atIndexNode.prevElement
+                    = new Node<>(atIndexNode.prevElement, value, atIndexNode);
+            atIndexNode.prevElement.prevElement.nextElement = atIndexNode.prevElement;
+            size++;
+            return;
+        }
+        throw new IndexOutOfBoundsException(index + " Out of bounds");
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T t : list) {
+            add(t);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexExtends(index);
+        return getNode(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNode(index);
+        T oldValue = node.element;
+        node.element = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (size == 0) {
```

Comment:
why not to use your indexExtends() method ? 
because current check doesn't cover situation where size = 3 and index = 8